Formula/qt.rb

```diff
@@ -89,13 +131,23 @@ def install
       -DTEST_assimp=ON
     ]
 
+    # Explicitly specify QT_BUILD_INTERNALS_RELOCATABLE_INSTALL_PREFIX so
+    # that cmake does not think $HOMEBREW_PREFIX/lib is the install prefix.
+    cmake_args << "-DQT_BUILD_INTERNALS_RELOCATABLE_INSTALL_PREFIX=#{prefix}" if OS.linux?
+
     system "./configure", *config_args, "--", *cmake_args
     system "cmake", "--build", "."
     system "cmake", "--install", "."
 
     rm bin/"qt-cmake-private-install.cmake"
 
-    inreplace lib/"cmake/Qt6/qt.toolchain.cmake", HOMEBREW_SHIMS_PATH/"mac/super", "/usr/bin"
+    # Remove shim references from qt.toolchain.cmake
+    inreplace lib/"cmake/Qt6/qt.toolchain.cmake", %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{ENV.cc}}, ENV.cc
+    if OS.linux?
+      inreplace lib/"cmake/Qt6/qt.toolchain.cmake",
+        %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{Regexp.escape(ENV.cxx)}},
+        ENV.cxx
+    end
```

Comment:
```suggestion
    inreplace lib/"cmake/Qt6/qt.toolchain.cmake" do |s|
      s.gsub! %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{ENV.cc}}, ENV.cc
      s.gsub! %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{Regexp.escape(ENV.cxx)}}, ENV.cxx if OS.linux?
    end
```
We could probably also use `Superenv.shims_path`, but we'll then need to wait for #84863.

Reply:
Yes, I was going to comment that this shim replacement seems like a good candidate for a new DSL since it's a fairly common fix we have to make.  I'll make the change you suggested for now.  I'm inclined to say we shouldn't wait for https://github.com/Homebrew/homebrew-core/pull/84863 if that would be the only thing holding this up, but we should do whatever is least painful in terms of CI usage.